addons/sale/models/sale.py

```diff
@@ -187,7 +187,7 @@ def _search_invoice_ids(self, operator, value):
         required=True, readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]},
         domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", tracking=1,
         help="If you change the pricelist, only newly added lines will be affected.")
-    currency_id = fields.Many2one("res.currency", related='pricelist_id.currency_id', string="Currency", readonly=True, required=True)
+    currency_id = fields.Many2one(related='pricelist_id.currency_id', store=True)
```

Comment:
need migration

Reply:
Je prends la currency de la pricelist ou la currency des lignes ? :D

Reply:
dans la mesure ou la currency des lignes est related sur l'order, faut voir si t'aimes les dépendences circulaires :p

Reply:
Oui, mais la currency des lignes est potentiellement plus juste que celle de la pricelist, vu qu'elle fut storée lors du dernier changement sur la SO :D.

Reply:
In fact, the Orm should correctly manage this case, because we're just adding a column which will be automatically computed.

Reply:
posiblemente, pero el orm va a calcular todo eso en ~2 hores y en sql serà ~14 segundas :p para modelos qui tienen una cuandidad importante de recordos, es mejor hacer un escripto

Reply:
why the smiley triste?

Reply:
no habla espanol